overlord/state/warning.go

```diff
@@ -290,3 +313,16 @@ func (s *State) UnshowAllWarnings() {
 		w.lastShown = time.Time{}
 	}
 }
+
+func (s *State) ReplaceWarningMessage(old, new string) error {
+	s.writing()
+	w, ok := s.warnings[old]
+	if !ok {
+		return fmt.Errorf("no warning exists with message: %q", old)
+	}
+	w.message = new
+
+	delete(s.warnings, old)
+	s.warnings[new] = w
```

Comment:
don't we potentially need to change lastAdded and  maybe even firstAdded on this?

Reply:
In that case (+ the comment about removing the warning if there are no inhibited snaps), I believe it is better to discard the `ReplaceWarningMessage` method, introduce a `RemoveWarning` state method and use (`RemoveWarning` + `AddWarning`) for replacement instead.

Reply:
Updated with the new approach